From 7e2766d3001fbb8923b095d6d37503d590187ba5 Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <jianzheng.zhou@freescale.com>
Date: Thu, 31 Oct 2013 16:49:50 +0800
Subject: [PATCH 29/32] ENGR00284562:check if tethering exist while start p2p
 dhcp server

If tethering already running , stop it and then start it.
Then it will update both ifaces msg to dnsmasq and will work ok.

Change-Id: I12ceccf3471c68455e5ae57d83d3832d6c13517d
Signed-off-by: Jianzheng Zhou <jianzheng.zhou@freescale.com>
---
 wifi/java/android/net/wifi/p2p/WifiP2pService.java | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/wifi/java/android/net/wifi/p2p/WifiP2pService.java b/wifi/java/android/net/wifi/p2p/WifiP2pService.java
index 8b07208..2914b37 100644
--- a/wifi/java/android/net/wifi/p2p/WifiP2pService.java
+++ b/wifi/java/android/net/wifi/p2p/WifiP2pService.java
@@ -2067,6 +2067,11 @@ public class WifiP2pService extends IWifiP2pManager.Stub {
             ifcg.setInterfaceUp();
             mNwService.setInterfaceConfig(intf, ifcg);
             /* This starts the dnsmasq server */
+            mNwService.tetherInterface(intf);
+            if (mNwService.isTetheringStarted()) {
+                if (DBG) logd("stop exist tethering and will restart it");
+                mNwService.stopTethering();
+            }
             mNwService.startTethering(DHCP_RANGE);
         } catch (Exception e) {
             loge("Error configuring interface " + intf + ", :" + e);
-- 
1.8.0

